app/src/main/java/com/duckduckgo/app/browser/BrowserChromeClient.kt

```diff
@@ -154,6 +154,64 @@ class BrowserChromeClient @Inject constructor(
         webViewClientListener?.closeCurrentTab()
     }
 
+    /**
+     * Called when a site's javascript tries to create a javascript alert dialog
+     * @return false to allow it to happen as normal; return true to suppress it from being shown
+     */
+    override fun onJsAlert(
+        view: WebView?,
+        url: String,
+        message: String,
+        result: JsResult,
+    ): Boolean {
+        if (webViewClientListener?.isActiveTab() == true) {
+            return false
+        }
+
+        Timber.v("onJsAlert called but is not the active tab; suppressing dialog")
+        result.cancel()
+        return true
```

Comment:
NIT: I would extract this code in a private function so you can reuse it in all callbacks.